src/Hedgehog/Gen.fs

```diff
@@ -474,10 +480,14 @@ module Gen =
     //
 
     let sampleTree (size : Size) (count : int) (g : Gen<'a>) : List<Tree<'a>> =
-        let seed = Seed.random ()
-        toRandom g
-        |> Random.replicate count
-        |> Random.run seed size
+        let action seed _ =
+            let (seed1, seed2) = Seed.split seed
+            (run seed1 size g), seed2
+
+        Seq.init count id
+        |> Seq.mapFold action (Seed.random ())
```

Comment:
(That's a funky way of avoiding to declare `let seed = Seed.random ()` isn't it?)